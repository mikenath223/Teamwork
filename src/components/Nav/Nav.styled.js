import styled from 'styled-components';
import { Button, Navbar } from 'react-bootstrap';
import { Link } from 'react-router-dom';
import { breakpoints } from 'styles/global';


export const Wrapper = styled(Navbar)`
  background-color: ${({ theme }) => theme.colors.white};
`;

export const LinkText = styled(Link)`
  margin-right: 25px;
  font-weight: ${({ theme }) => theme.fontWeight.bold};
  font-size: ${({ theme }) => theme.textSizes.large};

  &:hover {
  text-decoration: none;
  }

  &.nav-button {
    border: none;
    padding: 5px 8px;
    color: ${({ theme }) => theme.colors.white};
    background-color: ${({ theme }) => theme.colors.orange};
  }

  &.nav-button:hover {
    opacity: 0.7;
  }

  @media ${breakpoints.lg} {
    font-size: ${({ theme }) => theme.textSizes.regular};
  }
`;

export const Brand = styled(Navbar.Brand)`
  font-weight: ${({ theme }) => theme.fontWeight.bold};
  font-size: ${({ theme }) => theme.textSizes.larger};
  && {
    color: ${({ theme }) => theme.colors.orange};
  }
  &&:hover {
    color: ${({ theme }) => theme.colors.orange};
    opacity: 0.7;
  }

  @media ${breakpoints.lg} {
    font-size: ${({ theme }) => theme.textSizes.large};
  }
`;

export const NavButton = styled(Button)`
  background-color: ${({ theme }) => theme.colors.orange};
  
  &&:hover {
    opacity: 0.7;
    background-color: ${({ theme }) => theme.colors.orange}
  }
`;