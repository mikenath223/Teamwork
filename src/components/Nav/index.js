import React from 'react';
import { Navbar, Nav } from 'react-bootstrap';
import { 
  Wrapper,
  NavButton,
  Brand,
  LinkText
} from './Nav.styled'

const NavBar = () => (
    <Wrapper expand="md px-5">
      <Brand href="#home"><h2>teamwork</h2></Brand>
      <Navbar.Toggle aria-controls="basic-navbar-nav" />
      <Navbar.Collapse id="basic-navbar-nav">
        <Nav className="mx-auto">
          <LinkText href="#">Why teamwork?</LinkText>
          <LinkText href="#">Solutions?</LinkText>
          <LinkText href="#">Pricing?</LinkText>
          <LinkText href="#">Resources</LinkText>
        </Nav>
        <Nav className="mr-3">
          <LinkText href="#">Login</LinkText>
          <NavButton className="border border-0 btn-warning">Try Now</NavButton>
        </Nav>
      </Navbar.Collapse>
    </Wrapper>
);

export default NavBar;