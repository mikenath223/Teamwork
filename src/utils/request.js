import { auth, generateUserDocument, fire } from './firebaseConfig';
import db from './firebaseConfig';

/*
/ Util function to signup new user 
/ with email and password
*/
export const createUserWithEmailAndPassword = async (email, password, displayName, setError) => {
  try {
    const { user = {} } = await auth.createUserWithEmailAndPassword(email, password)
    await generateUserDocument(user, { displayName });
  } catch (error) {
    setError(error.message);
  }
};

/*
/ Util function to signIn user 
/ with email and password
*/
export const signInUserWithEmailAndPassword = (email, password, setError) => {
  auth.signInWithEmailAndPassword(email, password).catch(() => {
    setError("Error signing in with password and email!");
  });
}

/*
/ Util function to reset user password
*/
export const resetUserPassword = (email, setEmailHasBeenSent, setError) => {
  auth.sendPasswordResetEmail(email)
    .then(() => {
      setEmailHasBeenSent(true);
    })
    .catch(() => {
      setError("Error resetting password");
    });
}

/*
/ Util function to get users
*/
export const getUsers = async () => {
  try {
    let querySnapshot = [];
    await db.collection('users').get().then(query =>
      query.forEach(doc => querySnapshot.push({ id: doc.id, data: doc.data() })))
    return querySnapshot
  } catch (error) {
    console.log('Failed to retrieve users', JSON.stringify(error));
    throw new Error(error);
  }
};

/*
/ Util function to get chats
*/
export const getChats = async (recieverId, currentUserId, runIt) => {
  try {
    await new Promise(function (resolve) {
      return db.collection("chat").doc(recieverId)
        .onSnapshot(function (doc) {
          const chats = doc.data();
          if (chats.data) {
            const filteredChats = chats.data.filter(item =>
              item.recieverId === currentUserId);
            resolve(runIt(filteredChats));
          };
        });
    })
  } catch (error) {
    console.log('Chat uplink ended', JSON.stringify(error));
    throw new Error(error);
  }
}

/*
/ Util function to send chats
*/
export const sendFirestoreChat = (message, senderId, recieverId) => {
  try {
    db.collection('chat').doc(senderId).get().then((doc) => {
      if (doc.exists) {
        db.collection('chat').doc(senderId).update({
          data: fire.firestore.FieldValue.arrayUnion({
            message, recieverId, date: new Date().toLocaleString()
          })
        });
      } else {
        db.collection('chat').doc(senderId).set({
          data: [{
            message, recieverId, date: new Date().toLocaleString()
          }]
        });
      }
    })
  } catch (error) {
    console.log('Failed to send chat', JSON.stringify(error));
    throw new Error(error);
  }
};

/*
/ Util function to group chats
*/
export const runDispatch = (moment, dispatch, setChats,
  chats = [], count = 0) => (input) => {

    if (count >= 2) {
      count = 0;
      chats = [];
    }
    chats.push(...input);
    if (chats.length) {
      chats = chats.sort((a, b) =>
        new Date(a.date) - new Date(b.date))
      chats = chats.map(item => ({
        date: moment(item.date).fromNow(),
        message: item.message,
        recieverId: item.recieverId
      }));
      dispatch(setChats(chats));
    };
    count += 1;
  };

/*
/ Util function to save meetings
*/
export const sendFirebaseEvent = (event, currentUserId) => {
  db.collection('meetings').doc(currentUserId).get().then((doc) => {
    if (doc.exists) {
      db.collection('meetings').doc(currentUserId).update({
        data: fire.firestore.FieldValue.arrayUnion(event)
      });
    } else {
      db.collection('meetings').doc(currentUserId).set({
        data: [event]
      });
    }
  })
}