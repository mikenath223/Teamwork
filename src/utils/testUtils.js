import React from 'react';
import { ThemeProvider } from 'styled-components';
import { theme } from 'styles/global';
import { render } from '@testing-library/react';


export const withThemeProvider = (component) =>
  render(<ThemeProvider theme={theme}>{component}</ThemeProvider>);