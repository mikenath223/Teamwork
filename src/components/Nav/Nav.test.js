import React from 'react';
import 'jest-styled-components';
import { withThemeProvider } from 'utils/testUtils';
import Nav from './index';

describe('Nav', function () {
  it('should match snapshot', () => {
    const { container} = withThemeProvider(<Nav/>);

    expect(container).toMatchSnapshot();
  })
})