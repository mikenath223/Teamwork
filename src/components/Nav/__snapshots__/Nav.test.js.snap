// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should match snapshot 1`] = `
.c0 {
  background-color: #fff;
}

.c2 {
  margin-right: 15px;
  font-weight: 700;
  font-size: 18px;
}

.c1 {
  font-weight: 700;
  font-size: 35px;
}

.c1.c1 {
  color: #F58635;
}

.c1.c1:hover {
  color: #F58635;
  opacity: 0.7;
}

.c3 {
  background-color: #F58635;
}

.c3.c3:hover {
  opacity: 0.7;
  background-color: #F58635;
}

@media (max-width:992px) {
  .c2 {
    font-size: 14px;
  }
}

@media (max-width:992px) {
  .c1 {
    font-size: 18px;
  }
}

<div>
  <nav
    class="c0 navbar navbar-expand-md px-5 navbar-light"
  >
    <a
      class="c1 navbar-brand"
      href="#home"
    >
      teamwork
    </a>
    <button
      aria-controls="basic-navbar-nav"
      aria-label="Toggle navigation"
      class="navbar-toggler collapsed"
      type="button"
    >
      <span
        class="navbar-toggler-icon"
      />
    </button>
    <div
      class="navbar-collapse collapse"
      id="basic-navbar-nav"
    >
      <div
        class="mx-auto navbar-nav"
      >
        <a
          class="c2 nav-link"
          data-rb-event-key="#"
          href="#"
          role="button"
        >
          Why teamwork?
        </a>
        <a
          class="c2 nav-link"
          data-rb-event-key="#"
          href="#"
          role="button"
        >
          Solutions?
        </a>
        <a
          class="c2 nav-link"
          data-rb-event-key="#"
          href="#"
          role="button"
        >
          Pricing?
        </a>
        <a
          class="c2 nav-link"
          data-rb-event-key="#"
          href="#"
          role="button"
        >
          Resources
        </a>
      </div>
      <div
        class="mr-3 navbar-nav"
      >
        <a
          class="c2 nav-link"
          data-rb-event-key="#"
          href="#"
          role="button"
        >
          Login
        </a>
        <button
          class="c3 border border-0 btn-warning btn btn-primary"
          type="button"
        >
          Try Now
        </button>
      </div>
    </div>
  </nav>
</div>
`;
